pkg/cmdimpl/add.go

```diff
@@ -128,114 +121,26 @@ func updateParametersIfNecessary(gitClient git.Git) error {
 	return nil
 }
 
-func generateWegoSourceManifest() ([]byte, error) {
-	fluxRepoName, err := fluxops.GetRepoName()
-	if err != nil {
-		return nil, wrapError(err, "could not get flux repo name")
-	}
-
-	owner, err := getOwner()
-	if err != nil {
-		return nil, err
-	}
-
-	cmd := fmt.Sprintf(`create secret git "wego" \
-        --url="ssh://git@github.com/%s/%s" \
-        --private-key-file="%s" \
-        --namespace=%s`,
-		owner,
-		fluxRepoName,
-		params.PrivateKey,
-		params.Namespace)
-	if params.DryRun {
-		fmt.Printf(cmd + "\n")
-	} else {
-
-		_, err = fluxops.CallFlux(cmd)
+func getUrls(client git.Git, remote string) ([]string, error) {
+	if _, ok := client.(*git.GoGit); ok {
```

Comment:
If you do this you won't be able to mock it in the test. It will ignore it every time.

With this, you should be able to mock that

```
GitClient: &gitfakes.FakeGit{
	OpenStub: func(s string) (*gogit.Repository, error) {
		r, err := gogit.Init(memory.NewStorage(), memfs.New())
		Expect(err).ShouldNot(HaveOccurred())

		_, err = r.CreateRemote(&config.RemoteConfig{
			Name: "origin",
			URLs: []string{"http://foo/foo.git"},
		})
		Expect(err).ShouldNot(HaveOccurred())
		return r, nil
	},
},
```

Reply:
Without a remote, everything panicked. I had no way to fake the remote at the time so I upleveled it.  I'd be very happy to do something better later

Reply:
sounds good